src/site/stages/build/process-cms-exports/index.js

```diff
@@ -32,106 +165,29 @@ const entityAssemblerFactory = contentDir => {
    */
   const assembleEntityTree = (entity, ancestors = [], parentFieldName = '') => {
     // Handle circular references
-    const ancestorIds = ancestors.map(a => a.id);
-    const a = ancestors.find(r => r.id === toId(entity));
-    if (a) {
-      /* eslint-disable no-console */
-      // This logging is to help debug if AJV fails on an unexpected circular
-      // reference
-      console.log(`I'm my own grandpa! (${toId(entity)})`);
-      console.log(`  Parents:\n    ${ancestorIds.join('\n    ')}`);
-      /* eslint-enable no-console */
-
-      // NOTE: If we find a circular reference, it needs to be addressed in the
-      // transformer and accounted for in the transformed schema.
-      //
-      // If it isn't handled in the transformer, the post-transformation
-      // validation will fail because of a circular reference (AJV will throw
-      // up).
-      //
-      // If the modified child isn't accounted for in the transformed schema, it
-      // won't be valid (assuming we've omited a normally-required property to
-      // avoid the circular reference).
-      return a;
-    }
+    const a = findCircularReference(entity, ancestors);
+    if (a) return a;
 
-    // Pre-transformation JSON schema validation
-    const rawErrors = validateRawEntity(entity);
-    if (rawErrors.length) {
-      /* eslint-disable no-console */
-      console.warn(
-        chalk.yellow(
-          `${toId(entity)} (${getContentModelType(
-            entity,
-          )}) is invalid before transformation:`,
-        ),
-      );
-      console.warn(`${rawErrors.map(e => JSON.stringify(e, null, 2))}`);
-      rawErrors.forEach(e => {
-        console.warn(
-          chalk.yellow(`Data found at ${e.dataPath}:`),
-          JSON.stringify(get(entity, e.dataPath.slice(1))),
-        );
-      });
-      console.warn(`-------------------`);
-      /* eslint-enable no-console */
-
-      // Abort! (We may want to change this later)
-      throw new Error(`${toId(entity)} is invalid before transformation`);
-    }
+    validateInput(entity);
 
     const filteredEntity = getFilteredEntity(entity);
 
-    // Recursively expand entity references
-    for (const [key, prop] of Object.entries(filteredEntity)) {
-      // Properties with target_uuids are always arrays from tome-sync
-      if (Array.isArray(prop)) {
-        prop.forEach((item, index) => {
-          const { target_uuid: targetUuid, target_type: targetType } = item;
-
-          // We found a reference! Override it with the expanded entity.
-          if (targetUuid && targetType) {
-            filteredEntity[key][index] = assembleEntityTree(
-              readEntity(contentDir, targetType, targetUuid),
-              ancestors.concat([{ id: toId(entity), entity }]),
```

Comment:
Here we intentionally use the `entity` instead of `filteredEntity`...

Reply:
Ah, I bet if I took a closer look I might find that `filteredEntity` is just an alias for `entity`. :shrug: